server/backend/backend.go

```diff
@@ -646,19 +646,73 @@ func (b *Backend) DisableAutoMine() {
 	b.automine = false
 }
 
-func (b *Backend) GetTransactionResultByIndex(context.Context, flowgo.Identifier, uint32) (*access.TransactionResult, error) {
-	// TODO: implement
-	panic("GetTransactionResultByIndex not implemented")
+func (b *Backend) GetTransactionResultByIndex(ctx context.Context, id sdk.Identifier, index uint32) (*sdk.TransactionResult, error) {
+	results, err := b.GetTransactionResultsByBlockID(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	if uint32(len(results)) <= index {
+		return nil, status.Error(codes.NotFound, "TransactionResult not found")
+	}
+	return results[index], nil
 }
 
-func (b *Backend) GetTransactionsByBlockID(ctx context.Context, id flowgo.Identifier) ([]*flowgo.TransactionBody, error) {
-	// TODO: implement
-	panic("GetTransactionsByBlockID not implemented")
+func (b *Backend) GetTransactionsByBlockID(ctx context.Context, id sdk.Identifier) (result []*sdk.Transaction, err error) {
+	block, err := b.emulator.GetBlockByID(id)
+	if err != nil {
+		switch err.(type) {
+		case emulator.NotFoundError:
+			return nil, status.Error(codes.NotFound, err.Error())
+		default:
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+	}
```

Comment:
I believe this could be moved to GetBlockByID

Reply:
@sideninja  I checked this but it is a bit tricky, status.Error is GRPC error, I think it has been made as a middle ground ( instead of doing in adapter ) 

I can refactor all but feels a bit weird to import grpc on emulator layer.

Reply:
no no, you are right, I completely overlooked that's not the backend method but on emulator.